python/ray/serve/backend_worker.py

```diff
@@ -256,7 +257,39 @@ async def invoke_single(self, request_item: Query) -> Any:
 
         start = time.time()
         try:
-            result = await method_to_call(arg)
+            if self.config.internal_metadata.is_asgi_app:
+                app = self.callable._serve_asgi_app
+                req: Request = arg
+
+                class MockSender:
+                    def __init__(self) -> None:
+                        self.status_code: Optional[int] = 200
+                        self.header: Dict[str, str] = {}
+                        self.buffer: List[bytes] = []
+
+                    async def __call__(self, message):
+                        if (message["type"] == "http.response.body"):
+                            self.buffer.append(message["body"])
+                        if (message["type"] == "http.response.start"):
+                            self.status_code = message["status"]
+                            for key, value in message["headers"]:
+                                self.header[key.decode()] = value.decode()
+
+                    def build_starlette_response(
+                            self) -> starlette.responses.Response:
+                        return starlette.responses.Response(
+                            b"".join(self.buffer),
+                            status_code=self.status_code,
+                            headers=dict(self.header))
+
+                sender = MockSender()
+                path_params = req.scope.pop("path_params")
```

Comment:
If we *don't* copy the scope, we should leave a comment here explaining what the assumptions are that allow us to do that. This could lead to hard-to-debug errors in the future.